codalab/model/bundle_model.py

```diff
@@ -1000,22 +1002,37 @@ def update_bundle(self, bundle, update, connection=None):
         # Apply the column and metadata updates in memory and validate the result.
         metadata_update = update.pop('metadata', {})
         bundle.update_in_memory(update)
+
+        # Currently we only allow deleting fields in metadata dict using "DELETE" as the operation indicator.
+        # When the value of a metadata key is "DELETE", delete the key, value pair from the bundle.metadata dict
+        # Otherwise, update the (key, value) pair in the bundle.metadata dict
+        delete = False
         for (key, value) in metadata_update.items():
-            bundle.metadata.set_metadata_key(key, value)
+            if value == 'DELETE':
```

Comment:
Don't use 'DELETE' because the value could be 'DELETE'.  Either use `None` or have a separate parameter passed into this function which is just the set of keys to delete.

Reply:
@percyliang I chose to update the function parameter to have an additional `delete_keys`, as `None` is used as a default value for those fields, e.g.[request-queue](https://github.com/codalab/codalab-worksheets/blob/master/codalab/bundles/run_bundle.py#L32) and could be confusing for future references. The code and updated and tested against several test cases.

Reply:
Question: what goes wrong if we use `None` (which is a bit slicker)?  Is there a reason we need to distinguish between deleting and setting a metadata entry with a `None`?

Are there currently metadata entries in the database with a `None` value?